doc/tutorials/acrn-secure-boot-with-grub.rst

```diff
@@ -0,0 +1,179 @@
+.. _how-to-enable-acrn-secure-boot-with-grub:
+
+Enable ACRN Secure Boot with GRUB
+#################################
+
+This document shows how to enable ACRN secure boot with GRUB including:
+
+  -  ACRN Secure Boot Sequence
+  -  Generate GPG Key
+  -  Setup Standalone GRUB EFI Binary
+  -  UEFI Secure Boot
+
+  Validation Environment:
+	Hardware Platform: WHL-IPC-I7, Supported hardware described in :ref:`hardware`.
+	ACRN Scenario: Industry
+	Service VM: Ubuntu 18.04
+	GRUB: 2.02
+
+..note::
+	Note that GRUB may stop booting in case of problems, make sure you know how to recover a bootloader on your platform.
+
+ACRN Secure Boot Sequence
+*************************
+
+  ACRN can be booted by Multiboot compatible bootloader, following diagram illustrates the boot sequence of ACRN with GRUB:
+
+     .. image:: images/acrn_secureboot_flow.png
+        :align: center
+        :width: 600px
+
+
+  Refer to :ref:`using_grub` for details to enable GRUB on ACRN.
+
+  From point view of secureboot:
+	- UEFI firmware verifies Shim/GRUB
+	- GRUB verifies ACRN, Service VM kernel and pre-launched User VM kernel
+	- Service VM OS kernel verifies the Device Model (`acrn-dm`) and User VM OVMF bootloader (with the help of `acrn-dm`)
+	- User VM virtual bootloader (e.g. OVMF) starts the guest side verified boot process
+
+   This document provides guide to enable verification function of GRUB to verify ACRN binaries (such `acrn.bin`, Service VM kernel (`bzImage`) and pre-launched User VM kernel image if it is present).
+
+
+Generate GPG Key
+****************
+
+  GRUB supports loading of GPG signed files only if digital signatures enabled.
+  Example to generate GPG signing key::
```

Comment:
I _think_ you need a blank line here just before the code-block. Do you run `make doc` after your updates?